packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,222 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for merge-patch+json content-type")
+@scenarioService("/json-merge-patch")
+namespace Payload.JsonMergePatch;
+
+@doc("Details about a resource.")
+model Resource {
+  name: string;
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+}
+
+@doc("Details about a resource for patch operation.")
+model ResourcePatch {
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+}
+
+@doc("It is the model used by Resource model")
+model InnerModel {
+  name: string;
+  description?: string;
+}
+
+@scenario
+@scenarioDoc("""
+
+Expected input body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+""")
+@doc("Test content-type: application/merge-patch+json with required body")
+@route("/create/resource")
+@put
```

Comment:
What is the purpose of adding put operation?

Reply:
Simulate create operation, normally a resource needs to be created first, then update.